### PR TITLE
added delete column in db

### PR DESCRIPTION
**Pull request by C0smicG**
# Summary
Added operation field to DB schema.
Depending on the HTML request type it will be New, Update, or Delete:
Added Mock changelog data:
     - fixed bug with nullable values that were missed because the changelog didn't have many entries
## PR - Merge Checklist
- [ ] CR'd
- [x] README updated or N/A
- [ ] Fully tested and approved

### CODE REVIEW

File: internal/db/changelog.go [hunk 1]
```diff
@@ -93,8 +101,8 @@ func (c *Client) CreateChangeLogEntry(lastId int, entry *models.ChangelogEntry)
 	if err != nil {
 		return errors.Wrap(err, "Invalid Regex")
 	}
-
-	_, err = c.Conn.Exec("INSERT INTO changelog(rule_id,user_id,comment,created_at,response_code,enhanced_code,regex,priority,description,bounce_action) VALUES(?,?,?,?,?,?,?,?,?,?)", lastId, entry.UserID, entry.Comment, int32(time.Now().Unix()), entry.ResponseCode, entry.EnhancedCode, entry.Regex, entry.Priority, entry.Description, entry.BounceAction)
+	fmt.Println(entry.Operation)
```
Comment (xdsxc) [inline internal/db/changelog.go]:
```suggestion
```

**Review by xdsxc — COMMENTED** [review]
Found one small thing (debug statement), otherwise I think everything looks good.

Can you explain how this new column will be used?

**Comment by C0smicG** [comment]
When each CRUD route is hit it will add a special value for the changelog entry that will be created. (New, Update, or Delete). If and when we want to get the point of filtering the changelog we can use them to filter all the items that were deleted or create new

**Comment by xdsxc** [comment]
Gotcha, just confirming some assumptions. I suggest changing the datatype of `operation` in your table to an enum (see https://dev.mysql.com/doc/refman/8.0/en/enum.html)

- It decreases the amount of storage required in the DB for the column. For example, storing  "delete" in a column requires ~6 bytes. Storing the same value as an enum only requires 1 byte.
- It enforces database consistency. If you're new to the code base and you see that column you have no idea whats supposed to be there. This is ok for some columns (e.g. a "comment" column), but if the column is only supposed to have fixed values then seeing an enum tells the reader what values are expected to be there. 

You shouldn't have to do any special handling from the application perspective, the data will still be returned as a string.

**Review by xdsxc — APPROVED** [review]

**Pull request merged**